### DISCUSSION TITLE
Object of class Closure could not be converted to string

### DISCUSSION THREAD

**Discussion by sergioraldana** [filamentphp/filament#4157] [Help] [answered]
I'm trying to get a column value to use as extra attribute: 

`TextColumn::make('codigo')->extraAttributes(['class' => fn (Carrera $record): string => $record->color]),`

but I get this error: **Object of class Closure could not be converted to string**

**Comment by danharrin** [comment, marked as answer]
The closure needs to be passed to `extraAttributes()`, that will return the array of attributes, not just 1

**Reply by sergioraldana** [reply]
@danharrin Thanks!

`TextColumn::make('codigo')->extraAttributes(fn (Carrera $record): array => ['class' => $record->color]),`
